api/src/controllers/login.js

```diff
@@ -229,10 +251,64 @@ const getBranding = () => {
     });
 };
 
+const renderTokenLogin = (req, res) => {
+  return getBranding()
+    .then(branding => render('tokenLogin', req, branding, { tokenUrl: req.url }))
+    .then(body => res.send(body));
+};
+
+const createSessionRetry = (req, retry= 10) => {
+  return createSession(req).then(sessionRes => {
+    if (sessionRes.statusCode === 200) {
+      return sessionRes;
+    }
+
+    if (retry) {
+      return createSessionRetry(req, retry - 1);
```

Comment:
You don't have any built in timeout as far as I can see, so we don't have any control over how long this will take - it may run 10 iterations really quickly and still fail. I'd build in a 10ms or so delay so we know the loop waits at least 100ms. It also means the server gets a chance to breathe between attempts.